core/src/main/java/hudson/lifecycle/Lifecycle.java

```diff
@@ -75,6 +81,19 @@ public static synchronized Lifecycle get() {
                     NoClassDefFoundError x = new NoClassDefFoundError(e.getMessage());
                     x.initCause(e);
                     throw x;
+                } catch (InvocationTargetException e) {
+                    Throwable t = e.getCause();
+                    if (t instanceof RuntimeException) {
+                        throw (RuntimeException) t;
+                    } else if (t instanceof IOException) {
+                        throw new UncheckedIOException((IOException) t);
+                    } else if (t instanceof Exception) {
+                        throw new RuntimeException(t);
+                    } else if (t instanceof Error) {
+                        throw (Error) t;
+                    } else {
+                        throw new Error(e);
+                    }
```

Comment:
_Why checked exceptions were a terrible idea: Exhibit A_

Reply:
Indeed. I was tempted to use [Cy Young](https://www.launchableinc.com/a-tiny-library-for-java-exceptions-cy-young). That's probably what Kohsuke would have done back in the day. 😊 But I decided to keep it simple.

Reply:
This is probably a case when introducing a new Checked Exception throwing method AND deprecating the old one is reasonable. I do not intend to start a holywar here though :)

Reply:
In this case the appropriate code would really be more like

```java
try {
    ClassLoader cl = Jenkins.get().getPluginManager().uberClassLoader;
    instance = (Lifecycle)cl.loadClass(p).getDeclaredConstructor().newInstance();
} catch (Throwable t) {
    LOGGER.log(Level.WARNING, "Could not load requested ${hudson.lifecycle}", t);
    instance = new ExitLifecycle();
}
```